src/Computation/Values/Constants/ConstantSigned.cs

```diff
@@ -50,11 +50,56 @@ public ConstantDouble AsDouble()
         return ConstantDouble.Create(this);
     }
 
+    public ConstantSigned Divide(ConstantSigned value)
+    {
+        return Create(Size, _value / value._value);
+    }
+
     public ConstantSigned Extend(Bits size)
     {
         return new ConstantSigned(size, _value);
     }
 
+    public ConstantBool Greater(ConstantSigned value)
+    {
+        return new ConstantBool(_value > value._value);
+    }
+
+    public ConstantBool GreaterOrEqual(ConstantSigned value)
+    {
+        return new ConstantBool(_value >= value._value);
+    }
+
+    public ConstantBool Less(ConstantSigned value)
+    {
+        return new ConstantBool(_value < value._value);
+    }
+
+    public ConstantBool LessOrEqual(ConstantSigned value)
+    {
+        return new ConstantBool(_value <= value._value);
+    }
+
+    public ConstantSigned Remainder(ConstantSigned value)
+    {
+        return Create(Size, _value % value._value);
+    }
+
+    public ConstantSigned ShiftRight(ConstantUnsigned value)
+    {
+        return Create(Size, _value >> (int) (BigInteger) value);
```

Comment:
How does `BigInteger` determine when to do an arithmetic vs logical right shift?

Reply:
They are infinite/arbitrary width so there is no single 'high bit' to consider as the sign. It just works because in this class the value is interpreted as a signed integer, whereas in the other one it is interpreted as unsigned. So the result is always normalized to the appropriate value with respect to the size (which has to be stored separately alongside the raw `BigInteger`).